src/Knp/Rad/ResourceResolver/Caster/ServiceCaster.php

```diff
@@ -23,22 +23,16 @@ public function __construct(ContainerInterface $container)
     /**
      * {@inheritdoc}
      */
-    public function supports($value)
+    public function cast($value)
     {
-        if (false === is_string($value)) {
-            return false;
-        }
-
-        return 1 === preg_match('/^@(\w|\.)+/', $value);
+        return $this->container->get($value, ContainerInterface::NULL_ON_INVALID_REFERENCE);
     }
 
     /**
      * {@inheritdoc}
      */
-    public function cast($value)
+    public function supports($value)
     {
-        $service = $this->container->get($value, ContainerInterface::NULL_ON_INVALID_REFERENCE);
-
-        return null !== $service ? $service : $value;
+        return is_string($value) && 1 === preg_match('/^@(\w|\.)+/', $value);
```

Comment:
Maybe you can do it with the `strpos` function, It's a bit more performant than regex.